File: 1.29/solution.rb
require_relative "integrals.rb"

cube = -> (x) { x * x * x }

puts "\n\nSimple:\n\n"

[0.01, 0.001].each do |step|
  result = Integral.simple(cube, 0, 1, step)
  error = (0.25 - result).abs
  puts "Simple Cube #{step}: #{result}, err: #{error}"
end

puts "\n\nSimpson:\n\n"

[2, 10, 100, 1000, 5000, 200].sort.each do |steps|
  result = Integral.simpson(cube, 0, 1, steps)
  error = (result - 0.25).abs
  puts "Simpson Cube n=#{steps}: #{result}, err: #{error}"
end

File: 1.29/solution.rs
mod integrals;

fn main() {
    let step_array: [f64; 2] = [0.01, 0.001];
    let steps_array: [i32; 5] = [2, 10, 100, 200, 1000];
    let mut result: f64;

    println!("\n\nSimple:\n\n");
    
    for step in step_array.iter() {
        result = integrals::simple(0.0, 1.0, &cube, *step);
        println!("Simple Cube {}: {}, err: {}", step, result, (result - 0.25).abs());
    }

    println!("\n\nSimpson:\n\n");

    for steps in steps_array.iter() {
        result = integrals::simpson(0.0, 1.0, &cube, *steps);
        println!("Simpson Cube n={}: {}, err: {}", steps, result, (result - 0.25).abs());
    }
}

fn cube(x: f64) -> f64 {
    x * x * x
}